FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeProcessingJob.cs

```diff
@@ -569,8 +577,10 @@ private void AddFhirResourcesToCache(List<JObject> fhirResources)
                             }
 
                             // Upload to blob and log result
-                            var fileName = GetDataFileName(_inputData.DataEndTime, schemaType, _jobId, _outputFileIndexMap[schemaType]);
+                            var dateTime = _inputData.SplitParameters == null && _inputData.SplitParameters.ContainsKey(resourceType) ? _inputData.DataEndTime : _inputData.SplitParameters[resourceType].DataEndTime;
```

Comment:
use `_inputData.SplitParameters[resourceType].DataEndTime` only when `_inputData.SplitParameters != null` && `_inputData.SplitParameters.ContainsKey(resourceType)`?

Reply:
Fixed it, thanks!